### PR TITLE
Various minor changes

### PR DESCRIPTION
**Pull request by le717**
- Now PEP8 compliant
- Use [`with`](https://triangle717.wordpress.com/tutorials/python/with/) for reading files
- Support the Python 3 `print()` when running [under Python 2](https://triangle717.wordpress.com/tutorials/python/2a3printinput/#print) ([details](https://triangle717.wordpress.com/2015/02/11/python-future-module/))
- Make help text look a bit better
- Added many new code comments and two new docstrings
- Make `findFiles()` search the current directory as the first search
- Remove the unused `getCode()` method
- Make the part cache a private member of the object
- Set the `libraryLocation` and `startFile` defaults in the constructor (though IMO, there should be no defaults at all)
- Move the `log()` method right below the constructor (it's considered good practice to define your functions before calling them)
- Strip the `0` from comment lines (small thing before proper parsing is implemented)
- Use proper `parseOptLine()` method for parsing the optional lines

### CODE REVIEW

File: libldrparser.py [hunk 1]
```diff
@@ -125,13 +133,15 @@ def parsePart(self, line):
...
             filePath = self.findFile(myDef["partId"])
 
             if filePath is not None:
-                self.log("Caching Part: "+self.findFile(myDef["partId"]), 4)
-                f = open(filePath, 'r')
-                self._parts[myDef["partId"]] = self.buildPartData(f.read())
+                self.log("Caching Part: {0}".format(
+                         self.findFile(myDef["partId"])), 4)
+                with open(filePath, "r") as f:
+                    self.__parts[myDef["partId"]] = \
```
Comment (le717) [inline libldrparser.py:160]: Meh this line here to met the line length limit. :\

File: libldrparser.py [hunk 2]
```diff
@@ -204,35 +214,42 @@ def findFile(self, partPath):
...
+        if not locatedFile:
+            for path in paths:
+                if os.path.isfile(path):
+                    locatedFile = path
+                    break
 
         # Failing that, recursively search through every directory
         # in the library folder for the file.
         if not locatedFile:
-            for file in locate(os.path.basename(partPath),
```
Comment (le717) [inline libldrparser.py:215]: Technically, `file` is a reserved keyword in certain circumstances. That's why I changed it.

File: libldrparser.py [hunk 3]
```diff
@@ -204,35 +214,42 @@ def findFile(self, partPath):
...
 
-    def log(self, string, level=0):
-        if self.options["logLevel"] >= level:
-            print("[LDRParser] {0}".format(string))
+        @param {String} The part path to clean up.
+        @return {String}
+        """
+        return partName.lower().replace("\\",
+                                        os.path.sep).replace("/",
+                                                             os.path.sep)
```
Comment (le717) [inline libldrparser.py:269]: Again with the line length limit. :\

File: libldrparser.py (hunk 2 above)
Comment (JoshTheDerf) [inline libldrparser.py:215]: Good to know.

**Comment by JoshTheDerf** [comment]
Excellent once again! I am learning a lot just by reading through these. :)
Can't work on it today though. Kinda tied up at the moment.

**Pull request merged**